tutorials/2d/2d_movement.rst

```diff
@@ -0,0 +1,207 @@
+children: ``Sprite`` and ``CollisionShape2D``. You can use the Godot icon ("icon.png")
+for the Sprite's texture or use any other 2D image you have available.
+
+Open ``Project -> Project Settings`` and select the "Input Map" tab. Add the following
+input actions (see :ref:`InputEvent <doc_inputevent>` for details):
+
+.. image:: img/movement_inputs.png
+
+8-Way Movement
+--------------
+
+In this scenario, you want the user to press the four directional keys (up/left/down/right
+or W/A/S/D) and move in the selected direction. The name "8-way movement" comes from the
+fact that diagonal movement can be chosen by pressing two keys at once.
+
+.. image:: img/movement_8way.gif
+
+Add a script to the kinematic body and add the following code:
+
+::
+
+    extends KinematicBody2D
+
+    export (int) var speed = 200
+
+    var velocity = Vector2()
+
+    func get_input():
+        velocity = Vector2()
+        if Input.is_action_pressed('right'):
+            velocity.x += 1
+        if Input.is_action_pressed('left'):
+            velocity.x -= 1
+        if Input.is_action_pressed('down'):
+            velocity.y += 1
+        if Input.is_action_pressed('up'):
+            velocity.y -= 1
+        velocity = velocity.normalized() * speed
+
+    func _physics_process(delta):
+        get_input()
+        move_and_slide(velocity)
+
+In the ``get_input()`` function we check for the four key events and sum them
+up to get the velocity vector. This has the benefit of making two opposite keys
+cancel each other out, but will also result in diagonal movement being faster
+due to the two directions being added together.
+
+We can prevent that if we *normalize* the velocity, which means we set
+its *length* to ``1``, and multiply by the desired speed.
+
+.. tip:: If you've never used vector math before, or just need a refresher,
+         you can see an explanation of vector usage in Godot at :ref:`doc_vector_math`.
+
+
+
+
+
+Rotation + Movement
+-------------------
+
+This type of movement is sometimes called "Asteroids-style" because it resembles
+how that classic arcade game worked. Pressing left/right rotates the character,
+while up/down moves it forward or backward in whatever direction it's facing.
+
+.. image:: img/movement_rotate1.gif
+
+:: 
+    
+    extends KinematicBody2D
+
+    export (int) var speed = 200
+    export (float) var rot_speed = 1.5
+
+    var velocity = Vector2()
+    var rot_dir = 0
+
+    func get_input():
+        rot_dir = 0
+        velocity = Vector2()
+        if Input.is_action_pressed('right'):
+            rot_dir += 1
+        if Input.is_action_pressed('left'):
+            rot_dir -= 1
+        if Input.is_action_pressed('down'):
+            velocity = Vector2(-speed, 0).rotated(rotation)
+        if Input.is_action_pressed('up'):
+            velocity = Vector2(speed, 0).rotated(rotation)
+
+    func _physics_process(delta):
+        get_input()
+        rotation += rot_dir * rot_speed * delta
```

Comment:
Personal taste here but for educational purposes I'd rather use plain English variable names: rotation_direction etc. More as a rule of thumb for clarity, esp for beginners

Reply:
Yeah, I agree. I took this code from an older project and should have renamed them.